### DISCUSSION TITLE
Solidity formatting extension

### DISCUSSION THREAD

**Discussion by urosognjenovic** [Cyfrin/foundry-full-course-cu#1688] [General]
What formatting tools do you use for Solidity?

I am using Solidity by Nomic Foundation. I have also tried Solidity Visual Developer. None of these formats the variable type which would be highly appreciated. Is there something I am missing, maybe there's a way to configure these two to work with variable types?

**Comment by EngrPips** [comment]
Honestly, I just installed those two extensions you have, and it just works fine for me.

**Reply by urosognjenovic** [reply]
I gotta check what's wrong with my extensions then.

**Reply by EngrPips** [reply]
Yeah, You should.

**Reply by AbigirlMuchineripi** [reply]
hie @thebraveman17 ,What exactly is wrong with your extensions maybe we can figure it out together.

**Reply by FrankezeCode** [reply]
Bro Please I'm having the same issue, did you figure it ?

**Reply by EngrPips** [reply]
You can try what I suggested in the discussion you opened and also try closing and re-opening your `vscode`. Sometimes `vscode` misbehaves and needs to be reloaded.
